presto-main-base/src/main/java/com/facebook/presto/FullConnectorSession.java

```diff
@@ -48,17 +48,11 @@ public class FullConnectorSession
     private final SessionPropertyManager sessionPropertyManager;
     private final SqlFunctionProperties sqlFunctionProperties;
     private final Map<SqlFunctionId, SqlInvokedFunction> sessionFunctions;
+    private final RuntimeStats runtimeStats;
 
     public FullConnectorSession(Session session, ConnectorIdentity identity)
     {
-        this.session = requireNonNull(session, "session is null");
-        this.identity = requireNonNull(identity, "identity is null");
-        this.properties = null;
-        this.connectorId = null;
-        this.catalog = null;
-        this.sessionPropertyManager = null;
-        this.sqlFunctionProperties = session.getSqlFunctionProperties();
-        this.sessionFunctions = ImmutableMap.copyOf(session.getSessionFunctions());
+        this(builder(session, identity, null, null, null, null));
```

Comment:
Note: `sqlFunctionProperties` and `sessionFunctions` still have the same default values. I just moved that logic into the private constructor below.